jenkins-master/jobs/mozilla-aurora_functional/config.xml

```diff
@@ -89,24 +89,33 @@
       <selector class="hudson.plugins.copyartifact.StatusBuildSelector"/>
     </hudson.plugins.copyartifact.CopyArtifact>
     <hudson.plugins.xshell.XShellBuilder plugin="xshell@0.9-SNAPSHOT">
-      <commandLine>python runtests.py --type=functional --branch=mozilla-aurora --build-revision=$REVISION --build-locale=$LOCALE --installer-url=$INSTALLER_URL</commandLine>
+      <commandLine>python -u submission.py --test-type=functional --state=running --repository=mozilla-aurora --revision=$REVISION --locale=$LOCALE treeherder_venv</commandLine>
+      <executeFromWorkingDir>false</executeFromWorkingDir>
+    </hudson.plugins.xshell.XShellBuilder>
+    <hudson.plugins.xshell.XShellBuilder plugin="xshell@0.9-SNAPSHOT">
+      <commandLine>python runtests.py --type=functional --installer-url=$INSTALLER_URL</commandLine>
```

Comment:
To capture the result accurately, we'll need to keep track of both the mozharness exit code and the marionette exit code (or parse the test logs). For example, I think mozharness exits successfully as long as all the required steps complete; the exit code doesn't capture the test result. 

If the run-tests step in mozharness hits an output-timeout, does the mozharness exit code reflect that? I'm not sure it does.